src/segment/util.rs

```diff
@@ -13,48 +13,58 @@ use std::{path::Path, sync::Arc};
 #[derive(Debug)]
 pub struct SliceIndexes(pub usize, pub usize);
 
-pub fn load_block(
```

Comment:
Changing this will probably infer with what I'm doing a bit (and it seemingly already did), adding this in a new PR would be better.